### ISSUE TITLE
Live Wallpapers Lagging

### ISSUE THREAD

**Issue by qutewashington** [8bitPit/Niagara-Issues#604]
When i usw a normal Wallpaper in Niagara everything is great, but at the Moment i add a live Wallpaper everything is Lagging af. When i use another Launcher everything is great. Obly in Niagara.

Felix

**Comment by 8bitPit** [comment]
I'm sorry. Could you please provide a video showing the lag?

**Comment by qutewashington** [comment]
@8bitPit 
[Without Lag](http://sendvid.com/a388f19l)
[With Lag](http://sendvid.com/ov9fv2fe)

**Comment by 8bitPit** [comment]
Weird, the lice wallpaper shouldn't cause tremendous performance hiccups, can you also provide me your device specs / or your device's name?

**Comment by qutewashington** [comment]
@8bitPit
![](https://i.imgur.com/SlLYW1n.png)

**Comment by 8bitPit** [comment]
Sry, I didn't find any wallpaper related performance problems, since the wallpaper manager does the most job of rendering the background image.

**Comment by qutewashington** [comment]
@disacrol Google Pixel 2 live Wallpaper i Had No issues at any Other launcher

**Comment by disacrol** [comment]
Yes @qutewashington but with exactly which launchers did you try, other than the stock? And is that LWP available in the play store? The idea was to confirm this on a different device.

**Comment by qutewashington** [comment]
Nova Launcher, The normal Version

**Comment by disacrol** [comment]
I tried it out and I confirm the lag while using Pixel 2 LWP (your video is showing a beautiful beach from my country btw 😁).

However, I notice the very same amount of lag using any other launcher, not just Niagara. This is more clearly noticable while scrolling through apps in the typical app drawer most launcher3-based launchers have, specially if any kind of transparency is enabled (but lag is also evident even if it's not). Tried with Nova and with my Android One stock launcher.

My device is not a flagship by any means (SD 660, 6GB ram), but my guess is that, since Niagara animates the app scrolling with full transparency, it is expected of you have some heavy live wallpaper like the one in the video.

I tried for example that one with the Earth view and in that case lag was barely noticable.

Maybe Niagara could be improved in terms of how the apps scrolling is rendered, but at the moment it doesn't seem any worse than any other launcher out there.

TL; DR: Likely an expected side effect of the live wallpaper, depending on the specs of your device.

**Comment by Mvp1931** [comment]
I have same issue

**Comment by 8bitPit** [comment]
Does the issue still persist? If so, feel free to open it again.
